### PR TITLE
Checkpoint

### PR DESCRIPTION
**Pull request by unkrich**
## Issue

Opened as a checkpoint. Still issues, don't merge, but pls review

### CODE REVIEW

File: src/ingestion/collector.py [hunk 1]
```diff
@@ -9,73 +12,105 @@
...
 @dataclass
 class Collector:
     source: Source
-    filters: List[Any] # pure py function
     pipelines: List[Pipeline]
-    state: CollectorState = CollectorState()
+    configuration: CollectorConfiguration = CollectorConfiguration.default()
 
-    def _create_tasks(self):
+    def _create_tasks(self, discovered_data):
```
Comment (iporollo) [inline src/ingestion/collector.py:41]: is it necessary to pass `discovered_data` as a parameter here?

File: src/ingestion/collector.py [hunk 2]
```diff
@@ -9,73 +12,105 @@
...
-        if self.state.fetched_data is None:
-            return
-
-        [pipeline.push(self.state.fetched_data) for pipeline in self.pipelines]
-        self.state.fetched_data = None
+    def pass_data(self, data):
+        try:
+            [pipeline.push(data) for pipeline in self.pipelines]
+        except Exception as e:
+            print(e)
```
Comment (iporollo) [inline src/ingestion/collector.py:84]: we should log this

File: src/server/models.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ class Metric(DataClassDictMixin):
     time_window_end: datetime = field(default=None, metadata={"sa": Column(DateTime(timezone=True), nullable=False)})
     interval_length_sec: int = field(default=None, metadata={"sa": Column(Integer)})
 
-    id: int = field(default=None, metadata={"sa": Column(Integer, Sequence('metric_id_seq'), primary_key=True, autoincrement=True)})
+    id: str = field(default=None, metadata={"sa": Column(String(100), primary_key=True)})
```
Comment (iporollo) [inline src/server/models.py:74]: why change to a string based id?

File: src/ingestion/destinations/monosi.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    def __init__(self, configuration):
+        self.configuration = configuration
+        self.engine = None
+        self.connection = None
+
+    def _create_engine(self):
+        try:
+            return create_engine(self.configuration.connection_string())
+        except Exception as e:
+            raise e
```
Comment (iporollo) [inline src/ingestion/destinations/monosi.py:21]: If we are expecting this to bubble up to a higher level try-catch, we should just use `raise`

File: src/ingestion/sources/base.py [hunk 5]
```diff
@@ -246,7 +237,7 @@ def all(cls):
     @classmethod
     def default_for(cls, data_type): 
         data_type = data_type.lower()
-        if data_type == 'number':
+        if 'num' in data_type or 'int' in data_type:
```
Comment (iporollo) [inline src/ingestion/sources/base.py:240]: we should consider supporting other numeric types e.g. 

```
numeric_data_types = ['num', 'int', 'decimal', 'double', 'real', 'serial']

if any(substring in data_type for substring in numeric_data_types):
    ...

```

File: src/ui/src/pages/app/monitors/Index/components/MonitorsTable/index.tsx [hunk 6]
```diff
@@ -22,12 +22,23 @@ const MonitorsTable: React.FC<{
...
   const columns = [
     {
-      dataField: "metrics",
+      dataField: 'metrics',
       text: "Status",
       formatter: (cell: any, row: any) => {
-        return (
-          <span className="badge rounded-pill bg-success">{row.metrics} passed</span>
-        );
+      console.log(row)
```
Comment (iporollo) [inline src/ui/src/pages/app/monitors/Index/components/MonitorsTable/index.tsx:28]: remove

**Review by iporollo — DISMISSED** [review]
Overall LGTM, a few questions

File: src/ingestion/collector.py (hunk 1 above)
Comment (unkrich) [reply src/ingestion/collector.py:38]: here? No - in general? yes

File: src/server/models.py (hunk 3 above)
Comment (unkrich) [reply src/server/models.py:74]: not automatically created

File: src/ingestion/sources/base.py (hunk 5 above)
Comment (unkrich) [reply src/ingestion/sources/base.py:240]: Correct

File: src/ui/src/pages/app/monitors/Index/components/MonitorsTable/index.tsx (hunk 6 above)
Comment (unkrich) [reply src/ui/src/pages/app/monitors/Index/components/MonitorsTable/index.tsx]: Done

**Pull request merged**